great_expectations/datasource/generator/glob_reader_generator.py

```diff
@@ -119,7 +127,7 @@ def build_batch_kwargs_from_partition_id(self, generator_asset, partition_id=Non
                                         partition_id: partition_id
                                     })
         batch_kwargs = self._build_batch_kwargs_from_path(path[0], glob_config, reader_options=reader_options,
-                                                          limit=limit, partition_id=partition_id)
+                                                          limit=limit, reader_method=reader_method, partition_id=partition_id)
```

Comment:
```suggestion
                                                          limit=limit, partition_id=partition_id)
```

Reply:
I think between `self.reader_method` and `glob_config`, you've already got the relevant info here.